lib/Core/Memory.cpp

```diff
@@ -693,17 +760,66 @@ void PersistentState::commitPendingPersists() {
   cacheLineUpdates = pendingCacheLineUpdates;
 }
 
-ref<Expr> PersistentState::isPersisted() const {
+ref<Expr> PersistentState::isPersisted(ExecutionState &state) const {
+  /**
+   * This is extremely inefficient for large objects.
+   */
   // AND together all cache lines.
+  #if 0
   ref<Expr> result = isCacheLinePersisted(0);
   for (unsigned i = 1; i < numCacheLines(); ++i) {
     result = AndExpr::create(isCacheLinePersisted(i), result);
   }
   return EqExpr::create(getPersistedExpr(), result);
+  #else
+  // idxUnbounded->dump();
+  // idxBounded->dump();
+  state.constraints.addConstraint(idxConstraints.front());
```

Comment:
I will clean this up as this is the result of an attempted optimization that we no longer need. It is actually cleaned up in the Executor, but since we don't need it anymore, no reason to keep it.

I'll address the state passing in my general comments.